### PR TITLE
stop merging .kubeconfig files together

### LINKED ISSUE #4428: Merging .kubeconfig from multiple locations causes frustration and anger
The fact that kubectl merges config from multiple .kubeconfig locations was neat in theory, but in practice it's surprising, confusing, and occasionally infuriating. See #4426, #4397. If it causes this much pain for k8s devs, it's likely unusable by average kubernetes users.

Proposal: get rid of kubectl loading from a local .kubeconfig, `--kubeconfig` flag, and `$KUBECONFIG` env var . If a user needs to overrwrite behavior, they can use the `--cluster|context|user` flags. Having N ways to load config is just too confusing.

cc @deads2k, @satnam6502, @erictune

### PR DESCRIPTION
**Pull request by deads2k**
fixes #4428 

I'm running a vagrant e2e now.  Is there a jenkins job you can run to confirm e2e runs across other providers?

### CODE REVIEW

**Comment by deads2k** [comment]
@smarterclayton I'm pretty sure I'm already covered a Red Hat/Google agreement.  Do you know if there's some list I need to be on to make the googlebot happier?

**Comment by thockin** [comment]
I think the not is figuring everyone out.  Note that Googlers are off work
today.  :)

**Comment by zmerlynn** [comment]
Except the ever vigilant, slightly broken googlebot. It decided not to take
President's Day off.

**Comment by smarterclayton** [comment]
The next Bond movie is "Lawyers never sleep"

----- Original Message -----
> Except the ever vigilant, slightly broken googlebot. It decided not to take
> President's Day off.
> 
> On Mon, Feb 16, 2015 at 8:55 AM, Tim Hockin <notifications@github.com>
> wrote:
> 
> > I think the not is figuring everyone out. Note that Googlers are off work
> > today. :)

File: docs/kubeconfig-file.md [hunk 1]
```diff
@@ -93,10 +88,10 @@ kubectl config set property-name property-value
   Sets arbitrary value in .kubeconfig
 kubectl config unset property-name
   Unsets arbitrary value in .kubeconfig
-kubectl config view --local=true --global=false --kubeconfig=specific/filename --merged
-  Displays the merged (or not) result of the specified .kubeconfig file
+kubectl config view --kubeconfig=specific/filename
+  Displays the .kubeconfig file
 
---local, --global, and --kubeconfig are valid flags for all of these operations.
+--kubeconfig is a valid flags for all of these operations.
```
Comment (jlowdermilk) [inline docs/kubeconfig-file.md]: s / flags / flag

Comment (jlowdermilk) [inline docs/kubeconfig-file.md]: s / flags / flag

File: pkg/kubectl/cmd/cmd.go [hunk 2]
```diff
@@ -309,9 +309,8 @@ func (f *Factory) ClientMapperForCommand(cmd *cobra.Command) resource.ClientMapp
 //           3.  If the command line specifies one and the auth info specifies another, honor the command line technique.
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/cmd.go]: I think this comment can be simplified now that we are not merging.

File: pkg/kubectl/cmd/config/view.go [hunk 3]
```diff
@@ -39,20 +37,17 @@ func NewCmdConfigView(out io.Writer, pathOptions *pathOptions) *cobra.Command {
...
+		Long: `displays .kubeconfig settings or a specified .kubeconfig file.
 Examples:
-  // Show merged .kubeconfig settings.
-  $ kubectl config view
-
-  // Show only local ./.kubeconfig settings
-  $ kubectl config view --local`,
+  // Show settings from specified file
+  $ kubectl config view --kubeconfig=path/to/my/.kubeconfig`,
 		Run: func(cmd *cobra.Command, args []string) {
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/config/view.go]: Should have an example without --kubeconfig specified, since that's the default/most common use case.

**Comment by jlowdermilk** [comment]
Minor nits, otherwise LGTM.

**Comment by ghodss** [comment]
I would prefer this not get merged - I've written up my comments at https://github.com/GoogleCloudPlatform/kubernetes/issues/4428#issuecomment-74818745.

**Comment by deads2k** [comment]
Code comments addressed.  Leaving discussion of whether we should or shouldn't in #4428.

**Comment by jlowdermilk** [comment]
Closing this as abandoned, per decision in #4428.

**Pull request closed without merging**